### ISSUE TITLE
Randomly failing to sync filter headers in test cases

### ISSUE THREAD

**Issue by Christewart** [bitcoin-s/bitcoin-s#4976]
It seems my node gets hung up syncing filter headers. This is because (from the bitcoind perspective) we cannot find a filter that corresponds to what we requested with `getcfheaders`

```
2023-02-07 16:47:15,118 DEBUG [P2PClientActor] getcfheaders expects response from Peer(localhost:44202)
2023-02-07 16:47:15,118 DEBUG [P2PClientActor] received handled message NetworkMessage(NetworkHeader(RegTest,getcfheaders,UInt32Impl(37),ByteVector(4 bytes, 0x7d3d35ad)), GetCompactFilterHeadersMessage(Basic,UInt32Impl(150),DoubleSha256Digest(ed2adc662562bce01080233807e0c7d3e2968d6b11b0bacf21276ceb0a03a93c))) from Actor[akka://NeutrinoNodeWithWalletTest-1675788411246/deadLetters]
2023-02-07 16:47:15,118 DEBUG [P2PClientActor] Expecting response for getcfheaders for Peer(localhost:44202)
2023-02-07 16:47:15,118 INFO [P2PClientActor] P2PClient.sendNetworkMessage=getcfheaders
//no response after this
```

This is what i see inside of `regtest/debug.log` for my `bitcoind` node spun up in test cases.

```
2023-02-07T16:47:15Z [net] sending cfheaders (386 bytes) peer=3
2023-02-07T16:47:15Z [validation] UpdatedBlockTip: new block hash=55b7833feee2aa8418df496e115639ef5e3c579fe78f10644a3c8a02e7ca5922 fork block hash=52fadbb8c5c248b8a640bac1e0bd3eba94d34f2e697343e12555229d4f8c884b (in IBD=false)
2023-02-07T16:47:15Z [validation] BlockConnected: block hash=06c05483d59775bc7814528574d455b37ab25605c63369bb6159a855eb9784f0 block height=154
2023-02-07T16:47:15Z [leveldb] WriteBatch memory usage: db=txindex, before=0.0MiB, after=0.0MiB
2023-02-07T16:47:15Z [leveldb] WriteBatch memory usage: db=db, before=0.0MiB, after=0.0MiB
2023-02-07T16:47:15Z [net] received: getcfheaders (37 bytes) peer=3
2023-02-07T16:47:15Z [net] sending cfheaders (386 bytes) peer=3
2023-02-07T16:47:15Z [net] received: getcfheaders (37 bytes) peer=3
2023-02-07T16:47:15Z [net] sending cfheaders (386 bytes) peer=3
2023-02-07T16:47:15Z [net] received: getcfheaders (37 bytes) peer=3
2023-02-07T16:47:15Z [net] sending cfheaders (386 bytes) peer=3
2023-02-07T16:47:15Z [net] received: getcfheaders (37 bytes) peer=3
2023-02-07T16:47:15Z [net] sending cfheaders (386 bytes) peer=3
2023-02-07T16:47:15Z [default wallet] AddToWallet 18dff07e3ec9021e04fc654c22bf734dc9ee9e01dd07170d63b6176e497d79a0  new
2023-02-07T16:47:15Z [net] received: getcfheaders (37 bytes) peer=3
2023-02-07T16:47:15Z [net] sending cfheaders (386 bytes) peer=3
2023-02-07T16:47:15Z [net] received: getcfheaders (37 bytes) peer=3

2023-02-07T16:47:15Z [net] Failed to find block filter hashes in index: filter_type=basic, start_height=150, stop_hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed

2023-02-07T16:47:15Z [validation] UpdatedBlockTip: new block hash=06c05483d59775bc7814528574d455b37ab25605c63369bb6159a855eb9784f0 fork block hash=55b7833feee2aa8418df496e115639ef5e3c579fe78f10644a3c8a02e7ca5922 (in IBD=false)
2023-02-07T16:47:15Z [validation] BlockConnected: block hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed block height=155
2023-02-07T16:47:15Z [leveldb] WriteBatch memory usage: db=txindex, before=0.0MiB, after=0.0MiB
2023-02-07T16:47:15Z [leveldb] WriteBatch memory usage: db=db, before=0.0MiB, after=0.0MiB
2023-02-07T16:47:15Z [default wallet] AddToWallet 20b01b79f3589e1e33b22ef3cc4fa590f13a523ad490daed40dadd5c5f6f3b40  new
2023-02-07T16:47:15Z [validation] UpdatedBlockTip: new block hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed fork block hash=06c05483d59775bc7814528574d455b37ab25605c63369bb6159a855eb9784f0 (in IBD=false)
2023-02-07T16:47:15Z [http] Received a POST request for / from 127.0.0.1:63693
```

Here is the history of `stop_hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed`

```
2023-02-07T16:47:14Z [validation] NewPoWValidBlock: block hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed

2023-02-07T16:47:14Z [validation] BlockChecked: block hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed state=Valid

2023-02-07T16:47:14Z UpdateTip: new best=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed height=155 version=0x30000000 log2_work=8.285402 tx=167 date='2023-02-07T16:47:21Z' progress=1.000000 cache=0.0MiB(183txo)

2023-02-07T16:47:14Z [validation] Enqueuing BlockConnected: block hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed block height=155

2023-02-07T16:47:14Z [validation] Enqueuing UpdatedBlockTip: new block hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed fork block hash=06c05483d59775bc7814528574d455b37ab25605c63369bb6159a855eb9784f0 (in IBD=false)

2023-02-07T16:47:15Z [net] Failed to find block filter hashes in index: filter_type=basic, start_height=150, stop_hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed

2023-02-07T16:47:15Z [validation] BlockConnected: block hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed block height=155

2023-02-07T16:47:31Z [validation] UpdatedBlockTip: new block hash=3d7ab669dcaa6a23c1c166a71f3249ff3edef67d2eb5012628349f3573be607c fork block hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed (in IBD=false)

2023-02-07T16:47:31Z [validation] Enqueuing UpdatedBlockTip: new block hash=3d7ab669dcaa6a23c1c166a71f3249ff3edef67d2eb5012628349f3573be607c fork block hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed (in IBD=false)

2023-02-07T16:47:31Z [validation] UpdatedBlockTip: new block hash=3d7ab669dcaa6a23c1c166a71f3249ff3edef67d2eb5012628349f3573be607c fork block hash=3ca9030aeb6c2721cfbab0116b8d96e2d3c7e00738238010e0bc622566dc2aed (in IBD=false)
```

**Comment by Christewart** [comment]
I've attached my `application.log`/`debug.log` for future debugging purposes
[debug.log](https://github.com/bitcoin-s/bitcoin-s/files/10714042/debug.log)
[application.log](https://github.com/bitcoin-s/bitcoin-s/files/10714043/application.log)

**Comment by Christewart** [comment]
This is not fixed and is an underlying bug in bitcoin core, see https://github.com/bitcoin/bitcoin/issues/27085

**Comment by Christewart** [comment]
Workaround included in #5401
